### ISSUE TITLE
Download progress

### ISSUE THREAD

**Issue by ladislavklinc** [MasamMahmood/TestDownload#1]
Will check today or tomorrow and let you know if I can help...

**Comment by MasamMahmood** [comment]
@ladislavklinc Thank you.

**Comment by MasamMahmood** [comment]
@ladislavklinc Hope you doing well. Any update please?

**Comment by ladislavklinc** [comment]
I have a question regarding appointment details...in the current project you have multiple appointments but details are there only for one appointmet...I presume all appointments will have details right...

**Comment by MasamMahmood** [comment]
@ladislavklinc  Yes you are right. Every appointment has different appointment details.

**Comment by ladislavklinc** [comment]
One detail per appointment or multiple?

**Comment by MasamMahmood** [comment]
@ladislavklinc One detail per appointment. In this project I have used as a sample and every appointment is using same detail appointment. (JSON Files)
But in actual scenario every appointment has its own appointment detail.

**Comment by ladislavklinc** [comment]
Send me your email and I will send you what could be done...

**Comment by ladislavklinc** [comment]
https://share.getcloudapp.com/xQuW85zK (updated)

Here it is, you can see a first appointment and its downloads, where each download has its progress bar and whole appointment has progress bar of all downloads

**Comment by MasamMahmood** [comment]
@ladislavklinc  Thank you for your effort I really appreciate it. I have just checked it but one thing. 
According to your modified code. You divided One day four appointments into four tablecells but Its likely to be 
One date or day - all appointments in footer View or Child View. and then childview per appointment per progress should be merge or club of all files (which is in detail).

Attaching images . Mine one design is fine but everytime I start download it only executes last appointment not all. 

**In my case it starts only last appointment which is DEMET one but skips top three appointments and not started. check in Image. Actually I need to sort this out why it skipping top ones.**

Your code Image
<img width="367" alt="Screen Shot 2020-05-21 at 17 07 49" src="https://user-images.githubusercontent.com/36441313/82567912-b56d8d00-9b86-11ea-8551-2b295f69a267.png">

Mine Image. 

<img width="369" alt="Screen Shot 2020-05-21 at 17 09 48" src="https://user-images.githubusercontent.com/36441313/82567958-c74f3000-9b86-11ea-8e06-4c79da87cff4.png">

**Comment by ladislavklinc** [comment]
So each day is a cell and in current example we have 4 appointments

**Comment by MasamMahmood** [comment]
Yes exactly.

**Comment by ladislavklinc** [comment]
Ok, will check it tomorrow morning and see

**Comment by MasamMahmood** [comment]
Ok thank you. Hope you will get an idea.

**Comment by ladislavklinc** [comment]
[TestDownload.zip](https://github.com/MasamMahmood/TestDownload/files/4667406/TestDownload.zip)

**Comment by ladislavklinc** [comment]
now you have appointments per day, now you just have to implement the progress for the whole day...

**Comment by MasamMahmood** [comment]
@ladislavklinc Thank you.. Perfect. I was really struggling with that thing. 
Let me incorporate it with live API.  I have two questions regarding this.

1.  Appointment ID - link to Appointment Detail "appointmentId": 15473453, right if there is no appointment ID it shows me  **Nothing to download**  am I right?

2.  In case of error handling - ```files is not downloaded in any case can we track their file name of specific appointment. Appointment Detail JSON - textField (contains file name) by clicking appointment project name```. Please just help me in that as well I am really thankful for this I know you already helped me a lot. ```Error comes make it simple on click prints array of file names in console```. ( Rest of design I will do myself.)
 
<img width="369" alt="Screen Shot 2020-05-22 at 15 26 35" src="https://user-images.githubusercontent.com/36441313/82667599-dabed180-9c40-11ea-883a-9da3c0043941.png">

**Comment by ladislavklinc** [comment]
1) Check code in DownloadEntryCell, so for appointment that has 0 downloads I displayed this text just to let you know what was happening...

```
if appointmentDownload.downloadUrls.count == 0 {
  child_view.downloadLbl.text = "Nothing to download"
}
```

2) In AppointmentDownload class you can add another observer in **startDownload** method, so below downloadInfo.stateChangeBlock you can do:

```
downloadInfo.responsChangeBlock = {response in
  if let error = response.error {
    //we encountered an error
    print("Error")
  } else {
    print("Success")
  }
}
```

Then you can have logic to do whatever is needed in case you encounter an error

**Comment by MasamMahmood** [comment]
Thank you ...

**Comment by MasamMahmood** [comment]
@ladislavklinc Thank you

**Comment by MasamMahmood** [comment]
@ladislavklinc First of all thank you I really appreciate you spare time for me. 

I have created a new project and integrated with `live services` not static json data. However I have set overall progress and overall download label. 
Issues I have faced:

- when I select date from calendar download not started it directly goes to `Nothing to download`
even I have set a breakpoint files data is there but download not started. 

- Secondly - if I select any other date its added but download start of the first date selected only one appointment. not complete one and stuck on downloading.
- Thirdly - if I select same date added `superview` again but download not started. 

I have followed the same procedure which you have coded and mentioned. 
I use `18May2020` (have two appointments)  and `19May2020` (one appointment) dates. 
**Note: Files are `DownloadViewController` and your file `AppointmentDownload.swift`**

Attaching project with Live Data.

[DMTest.zip](https://github.com/MasamMahmood/TestDownload/files/4708371/DMTest.zip)



<img width="365" alt="Screen Shot 2020-05-31 at 23 22 48" src="https://user-images.githubusercontent.com/36441313/83362157-c2b12580-a397-11ea-85f5-ffe40f0f6b2a.png">
